joomla/joomla-cms
# Category Manager :: Wrong message after publishing sub-category

Solved issue from #5596

**The issue:**

If you have a category and a related sub-category and unpublish the category, both get unpublished. If you now publish the sub-category, the message will be "Successful" but the sub-category hasn't been published. This message should be changed.

**The solution:**
1. I needed to create a new language constant, because there was none that could be used. Therefore I created the constant, COM_CATEGORIES_PUBLISHING_NOT_SUCCESSFUL="Publishing was not successful."
2. If a sub-category gets published, there is a check that checks for the related main category and if this one is active. If not, an error is set. The problem was, that there was no check if an error occurred during this process and therefor the error message always said that everything was fine and published. I created a new variable $errors = $model->getErrors(); which will inherit all the errors that are set. After that I also needed to create the check and throw an error.

**How to test:**
- Navigate to Category Manager
- Create a main category
- Create a sub-category that is related to the new main category
- unpublish the main category
- Both main & sub-category will get unpublished
- Click on the sub-category and try to publish and check the message

**Expected Result after solution.** 
An error message will be displayed that the publishing was not successful. 

_Worked as a group on that issue: @icampus, @FPerisa, @flow87 and @xsability_
